lib/dalli/protocol/binary.rb

```diff
@@ -18,11 +18,16 @@ def response_processor
 
       private
 
-      # Retrieval Commands
-      def get(key, options = nil)
-        req = RequestFormatter.standard_request(opkey: :get, key: key)
+      def safe_get(key, options = nil)
+        req = RequestFormatter.standard_request(opkey: :getk, key: key)
         write(req)
-        response_processor.get(cache_nils: cache_nils?(options))
+        response_processor.getk(key, cache_nils: cache_nils?(options)).last
+      rescue Dalli::SocketCorruptionError => e
+        Dalli.logger.debug { e.inspect }
+        
+        # Close connection; connection manager logic will automatically reopen it after a delay.
+        # This raises its own network error handled upstream.
+        down!   
```

Comment:
I think `error_on_request!` or `reconnect!` would be preferable over `down!` since `down!` tracks `down_at` and could have longer impact. https://github.com/aha-app/dalli/blob/e692421ebc923144a98716f079ac5e4d172815a0/lib/dalli/protocol/connection_manager.rb#L178-L195

Reply:
My thinking was that I wanted it to respect the `down_retry_delay` argument to be more compatible upstream. Not really a concern anymore. 

And actually this does look similar to where `error_on_request!` gets used, so seems appropriate.